packages/sdk-metrics/src/aggregator/ExponentialHistogram.ts

```diff
@@ -0,0 +1,606 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Accumulation,
+  AccumulationRecord,
+  Aggregator,
+  AggregatorKind,
+} from './types';
+import {
+  DataPointType,
+  ExponentialHistogramMetricData,
+} from '../export/MetricData';
+import { diag, HrTime } from '@opentelemetry/api';
+import { InstrumentDescriptor, InstrumentType } from '../InstrumentDescriptor';
+import { Maybe } from '../utils';
+import { AggregationTemporality } from '../export/AggregationTemporality';
+import { Buckets } from './exponential-histogram/Buckets';
+import { getMapping } from './exponential-histogram/mapping/getMapping';
+import { Mapping } from './exponential-histogram/mapping/types';
+import * as util from './exponential-histogram//util';
+
+/**
+ * Internal value type for ExponentialHistogramAggregation.
+ * Differs from the exported type as undefined sum/min/max complicate arithmetic
+ * performed by this aggregation, but are required to be undefined in the exported types.
+ */
+interface InternalHistogram {
+  positive: {
+    offset: number;
+    bucketCounts: number[];
+  };
+  negative: {
+    offset: number;
+    bucketCounts: number[];
+  };
+  hasMinMax: boolean;
+  min: number;
+  max: number;
+  sum: number;
+  count: number;
+  scale: number;
+  zeroCount: number;
+}
+
+// HighLow is a utility class used for computing a common scale for
+// two exponential histogram accumulations
+class HighLow {
+  static combine(h1: HighLow, h2: HighLow): HighLow {
+    return new HighLow(Math.min(h1.low, h2.low), Math.max(h1.high, h2.high));
+  }
+  constructor(public low: number, public high: number) {}
+}
+
+const MAX_SCALE = 20;
+const DEFAULT_MAX_SIZE = 160;
+const MIN_MAX_SIZE = 2;
```

Comment:
Where do these values come from? Is this spec or just some reasonable defaults?

Reply:
`MaxScale` and `MaxSize` are from the spec: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#base2-exponential-bucket-histogram-aggregation. `MinMaxSize` is a consequence of the minimum scale of -10. At scale -10 the entire floating point range can fit into 2 buckets, thus we need a minimum size of 2.